### PR TITLE
Make it more modular

### PR DESCRIPTION
**Pull request by jimCresswell**
Make the code more modular and add a unit test...

### CODE REVIEW

**Comment by jimCresswell** [comment]
This makes the code more modular, and I've added one unit test which runs in team city, but is not part of the coverage test run... are we running the tests once for the results and once for coverage? Either way I'd like to merge this so we can iterate and start adding unit tests.

File: lib/specifications/files/feature-files/tags.js [hunk 1]
```diff
@@ -3,6 +3,12 @@
  */
 'use strict';
 
+module.exports = {
+  count: count,
+  filter: filter
```
Comment (sponte) [inline lib/specifications/files/feature-files/tags.js]: Filter file objects or similar.

File: lib/specifications/files/process-files.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+   */
+  return function filePathToFileObject(filePath) {
+    var file = {};
+
+    file.name = path.basename(filePath);
+    file.filePath = filePath;
+
+    file.route = path.posix.join(projectRoute, projectData.repoName, filePath);
+
+    file.isFeatureFile = /.*\.feature/.test(filePath);
```
Comment (sponte) [inline lib/specifications/files/process-files.js]: add $

File: lib/specifications/files/process-files.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+function processFileContent(file) {
+  var fileContent = file.content;
+
+  if (!fileContent || !fileContent.length) {
+    file.empty = true;
+  }
+
+  try {
+    if (file.isFeatureFile) {
+      file.data = Parser.parse(fileContent);
```
Comment (sponte) [inline lib/specifications/files/process-files.js]: Feature parser

File: lib/specifications/projects/views.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+/**
```
Comment (sponte) [inline lib/specifications/projects/views.js]: Some other name, "projection"?

File: lib/specifications/repository-types/git/git-fetching.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
+/**
```
Comment (sponte) [inline lib/specifications/repository-types/git/git-fetching.js]: don't repeat git

File: lib/specifications/projects/views.js (hunk 4 above)
Comment (jimCresswell) [inline lib/specifications/projects/views.js]: This is kind of a pain to rename... how do you feel about calling these project views and renaming the views folder to `templates`?

Comment (sponte) [inline lib/specifications/projects/views.js]: Sounds good

Comment (jimCresswell) [inline lib/specifications/projects/views.js]: I've gone off that idea because Express uses the term 'views'...

**Comment by jimCresswell** [comment]
@sponte okay, done. I changed views to "project views", not ideal, but they really are views...

On a side note, if the CI fails take a look in the cucumber.json artefact, it contains the stack trace for some entirely unsatisfactory reason.

**Comment by jimCresswell** [comment]
@sponte changes made, tests passed...

**Comment by jimCresswell** [comment]
Once this is deployed we'll have to change the specs.json file in the other project.

**Pull request merged**